### PR TITLE
Fixed rendezvous handling in SRT pre-configure

### LINKED ISSUE #1195: [BUG] URI Parser issues with rendezvous mode
The following URI request fails to set the correct connection mode:
```
"srt://192.168.1.91:4200?adapter=192.168.1.237&port=4200&mode=rendezvous"
```
Does not work with the latest master 7fda3652919da9ba8ff64b9f8d8cebe74ed7c0f1
Works with SRT v1.4.1.

**srt-test-live logs**
```
./srt-test-live "srt://192.168.1.91:4200?adapter=192.168.1.237&port=4200&mode=rendezvous" udp://192.168.1.237:5200 -v
SRT parameters specified:
	adapter = '192.168.1.237'
	mode = 'rendezvous'
	port = '4200'
Opening SRT source rendezvous(blocking, backlog=1) on 192.168.1.91:4200
WARNING: failed to set options: 
mode, 

ERROR (app): ConfigurePre
Open FAILED - closing SRT sockets
MEDIA CREATION FAILED: ConfigurePre - exitting.
```

### PR DESCRIPTION
**Pull request by ethouris**
Fixes #1195

### CODE REVIEW

**Comment by maxsharabayko** [comment]
This way rendezvous also was not working, although it should according to the spec.
It does not look like this PR fixes this.
```
./srt-test-live "srt://192.168.178.182:4200?adapter=192.168.1.237" udp://192.168.1.237:5200 -v
```

**Comment by ethouris** [comment]
Weirdly, mode interpretation happens in two places and in both places differently. This also needs to be fixed.

File: apps/socketoptions.cpp [hunk 1]
```diff
@@ -22,28 +22,28 @@ extern const std::map<std::string, int> enummap_transtype = {
...
-    string modestr = "default";
+const char* const SocketOption::mode_names[3] = {
+    "listener", "caller", "rendezvous"
+};
 
-    if ( options.count("mode") )
-    {
-        modestr = options["mode"];
-    }
+SocketOption::Mode SrtInterpretMode(string modestr, string host, string adapter)
```
Comment (maxsharabayko) [inline apps/socketoptions.cpp:30]:
```suggestion
SocketOption::Mode SrtInterpretMode(const string& modestr, const string& host, const string& adapter)
```

File: apps/transmitmedia.cpp [hunk 2]
```diff
@@ -440,10 +438,12 @@ void SrtCommon::OpenRendezvous(string adapter, string host, int port)
     if ( stat == SRT_ERROR )
         Error("ConfigurePre");
 
-    sockaddr_in localsa = CreateAddrInet(adapter, port);
+    int outport = m_outgoing_port ? m_outgoing_port : port;
```
Comment (maxsharabayko) [inline apps/transmitmedia.cpp:441]:
```suggestion
    const int outport = m_outgoing_port ? m_outgoing_port : port;
```

File: testing/testmedia.cpp [hunk 3]
```diff
@@ -265,33 +265,31 @@ void SrtCommon::InitParameters(string host, string path, map<string,string> par)
...
-    if (m_mode == "client")
-        m_mode = "caller";
-    else if (m_mode == "server")
-        m_mode = "listener";
-
-    if (m_mode == "listener" && !m_group_nodes.empty())
+    if (!m_group_nodes.empty() && mode != SocketOption::CALLER)
     {
-        Error("Multiple nodes (redundant links) only supported in CALLER (client) mode.");
+        Error("Group node specification only available in caller mode");
```
Comment (maxsharabayko) [inline testing/testmedia.cpp:287]:
```suggestion
        Error("Group node specification is only available in caller mode");
```

File: testing/testmedia.cpp [hunk 4]
```diff
@@ -1013,9 +1011,11 @@ void SrtCommon::SetupRendezvous(string adapter, int port)
     bool yes = true;
     srt_setsockopt(m_sock, 0, SRTO_RENDEZVOUS, &yes, sizeof yes);
 
-    sockaddr_in localsa = CreateAddrInet(adapter, port);
+    int outport = m_outgoing_port ? m_outgoing_port : port;
```
Comment (maxsharabayko) [inline testing/testmedia.cpp:1014]:
```suggestion
    const int outport = m_outgoing_port ? m_outgoing_port : port;
```

**Pull request merged**